### PR TITLE
[CARBONDATA-1312] fix list partition compare issue

### PR DESCRIPTION
**Pull request by lionelcao**
need to convert string to bytes in list partition comparator

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3111/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/520/

File: core/src/main/java/org/apache/carbondata/core/scan/filter/partition/PartitionFilterUtil.java [hunk 1]
```diff
@@ -63,6 +64,9 @@ public static BitSet getPartitionMapForRangeFilter(PartitionInfo partitionInfo,
           for (String value : values.get(i)) {
             Object listValue = PartitionUtil.getDataBasedOnDataType(value, partitionColumnDataType,
                 timestampFormatter, dateFormatter);
+            if (listValue instanceof String) {
+              listValue = ByteUtil.toBytes((String)listValue);
+            }
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/partition/PartitionFilterUtil.java]: please improve performance and refractory code.
we should convert value only once. please move the code to the outside of for loop.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/partition/TestQueryForPartitionTable.scala [hunk 2]
```diff
@@ -77,8 +61,36 @@ class TestQueryForPartitionTable  extends QueryTest with BeforeAndAfterAll {
...
+  test("detail query on partition table: hash table") {
+    // EqualTo
+    checkAnswer(sql("select empno, empname, designation, doj, workgroupcategory, workgroupcategoryname, deptno, deptname, projectcode, projectjoindate, projectenddate, attendance, utilization, salary from hashTable where empno = 13"),
+      sql("select  empno, empname, designation, doj, workgroupcategory, workgroupcategoryname, deptno, deptname, projectcode, projectjoindate, projectenddate, attendance, utilization, salary from originTable where empno = 13"))
+    // In
+    checkAnswer(sql("select empno, empname, designation, doj, workgroupcategory, workgroupcategoryname, deptno, deptname, projectcode, projectjoindate, projectenddate, attendance, utilization, salary from hashTable where empno in (11, 13)"),
+      sql("select  empno, empname, designation, doj, workgroupcategory, workgroupcategoryname, deptno, deptname, projectcode, projectjoindate, projectenddate, attendance, utilization, salary from originTable where empno in (11, 13)"))
+  }
+
+  test("detail query on partition table: range partition") {
```
Comment (QiangCai) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/partition/TestQueryForPartitionTable.scala]: please add test case for list and range partition:
create table:
1. partition column is string, and add partition column to dictionary_exclude (means )
2. partition column is string
query:
filter on partition column , support <. <=, >, >= operate.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3120/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/529/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/542/

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3134/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/544/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3136/

**Comment by QiangCai** [comment]
LGTM

File: core/src/main/java/org/apache/carbondata/core/scan/filter/partition/PartitionFilterUtil.java (hunk 1 above)
Comment (lionelcao) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/partition/PartitionFilterUtil.java]: cannot move to outside here.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/partition/TestQueryForPartitionTable.scala (hunk 2 above)
Comment (lionelcao) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/partition/TestQueryForPartitionTable.scala]: test case added.

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**